From a16d20f51c12780e4ce6d7dc76c06dba6378d751 Mon Sep 17 00:00:00 2001
From: Hikari-no-Tenshi <kyryljan.serhij@gmail.com>
Date: Thu, 10 Nov 2016 01:22:51 +0200
Subject: [PATCH] libstagefright: fix camera/encoder for targets that using native handle source

Some devices (e.g. Xiaomi Mi4c) using native handle source, activated by
flag TARGET_USES_MEDIA_EXTENSIONS from hardware/qcom/media-caf.

Change-Id: Ifbf477218bca9b56a8f1cce2601ebc4f6a37fee4
---

diff --git a/media/libstagefright/Android.mk b/media/libstagefright/Android.mk
index c605e1b..15a5160 100644
--- a/media/libstagefright/Android.mk
+++ b/media/libstagefright/Android.mk
@@ -128,9 +128,19 @@
         libdl \
         libRScpp \
 
+ifeq ($(BOARD_USE_SAMSUNG_CAMERAFORMAT_NV21), true)
+# This needs flag requires the following string constant in
+# CameraParametersExtra.h:
+#
+# const char CameraParameters::PIXEL_FORMAT_YUV420SP_NV21[] = "nv21";
+LOCAL_CFLAGS += -DUSE_SAMSUNG_CAMERAFORMAT_NV21
+endif
+
+ifneq ($(TARGET_USES_MEDIA_EXTENSIONS),true)
 ifeq ($(TARGET_HAS_LEGACY_CAMERA_HAL1),true)
 LOCAL_CFLAGS += -DCAMCORDER_GRALLOC_SOURCE
 endif
+endif
 
 LOCAL_CFLAGS += -Wno-multichar -Werror -Wno-error=deprecated-declarations -Wall
 
